Documentation/design/building-your-csv.md

```diff
@@ -0,0 +1,250 @@
+
+The namespace is used when a CSV will remain private to a specific namespace. Only users of that namespace will be able to view or instantiate the Operator. If you plan on distributing your Operator to many namespaces or clusters, you may want to explore bundling it into a [Catalog](architecture.md#catalog-registry-design).
+
+The namespace listed in the CSV within a catalog is actually a placeholder, so it is common to simply list `placeholder`. Otherwise, loading a CSV directly into a namespace requires that namespace, of course.
+
+```yaml
+apiVersion: operators.coreos.com/v1alpha1
+kind: ClusterServiceVersion
+metadata:
+  name: mongodboperator.v0.3
+  namespace: placeholder
+```
+
+## Your Custom Resource Definitions
+There are two types of CRDs that your Operator may use, ones that are “owned” by it and ones that it depends on, which are “required”.
+### Owned CRDs
+
+The CRDs owned by your Operator are the most important part of your CSV. This establishes the link between your Operator and the required RBAC rules, dependency management and other under-the-hood Kubernetes concepts.
+
+It’s common for your Operator to use multiple CRDs to link together concepts, such as top-level database configuration in one object and a representation of replica sets in another. List out each one in the CSV file.
+
+**DisplayName**: A human readable version of your CRD name, eg. “MongoDB Standalone”
+
+**Description**: A short description of how this CRD is used by the Operator or a description of the functionality provided by the CRD.
+
+**Group**: The API group that this CRD belongs to, eg. database.example.com
+
+**Kind**: The machine readable name of your CRD
+
+**Name**: The full name of your CRD
+
+The next two sections require more explanation. 
+
+**Resources**:
+Your CRDs will own one or more types of Kubernetes objects. These are listed in the resources section to inform your end-users of the objects they might need to troubleshoot or how to connect to the application, such as the Service or Ingress rule that exposes a database.
+
+It’s recommended to only list out the objects that are important to a human, not an exhaustive list of everything you orchestrate. For example, ConfigMaps that store internal state that shouldn’t be modified by a user shouldn’t appear here.
+
+**SpecDescriptors**:
+
+These are a way to hint UIs with certain inputs or outputs of your Operator that are most important to an end users. If your CRD contains the name of a Secret or ConfigMap that the user must provide, you can specify that here. These items will be linked to and highlighted in compatible UIs. The description allows for you to explain how this is used by the Operator.
+
+todo(?): list out the options
+
+Below is an example of a MongoDB “standalone” CRD that requires some user input in the form of a Secret and ConfigMap, and orchestrates Services, StatefulSets, Pods and ConfigMaps.
+
+```yaml
+      - description: >-
```

Comment:
github doesn't like the `>-` (makes the rest of the yaml not syntax highlight)

Reply:
Quick google yields nothing. We can move it to the bottom at least?